packages/drivers/routerlicious-driver/src/deltaStorageService.ts

```diff
@@ -18,37 +18,35 @@ export class DocumentDeltaStorageService implements IDocumentDeltaStorageService
     constructor(
         private readonly tenantId: string,
         private readonly id: string,
-        private readonly tokenProvider: api.ITokenProvider,
         private readonly storageService: IDeltaStorageService) {
     }
 
     public async get(from?: number, to?: number): Promise<api.ISequencedDocumentMessage[]> {
-        return this.storageService.get(this.tenantId, this.id, this.tokenProvider, from, to);
+        return this.storageService.get(this.tenantId, this.id, from, to);
     }
 }
 
 /**
  * Provides access to the underlying delta storage on the server for routerlicious driver.
  */
 export class DeltaStorageService implements IDeltaStorageService {
-    constructor(private readonly url: string) {
+    constructor(private readonly url: string, private readonly tokenProvider: ITokenProvider) {
     }
 
     public async get(
         tenantId: string,
         id: string,
-        tokenProvider: api.ITokenProvider,
         from?: number,
         to?: number): Promise<api.ISequencedDocumentMessage[]> {
         const query = querystring.stringify({ from, to });
 
         let headers: { Authorization: string } | null = null;
 
-        const token = (tokenProvider as TokenProvider).token;
+        const storageToken = await this.tokenProvider.fetchStorageToken();
 
-        if (token) {
+        if (storageToken) {
```

Comment:
You say in another comment "We should remove the undefined check and enforce token for all client calls."

Reply:
Correct! I was mostly incorporating changes in this PR. Will do the strict token check in a separate PR.